tests/ui/crashes/ice-3717.fixed

```diff
@@ -0,0 +1,11 @@
+#![deny(clippy::implicit_hasher)]
+
+use std::collections::HashSet;
+
+fn main() {}
+
+pub fn ice_3717<S: ::std::hash::BuildHasher + Default>(_: &HashSet<usize, S>) {
+    //~^ ERROR: parameter of type `HashSet` should be generalized over different hashers
+    let _ = [0u8; 0];
+    let _: HashSet<usize> = HashSet::default();
+}
```

Comment:
When I run the test in local, the same file is output as when it was deleted.
I need to investigate this a bit more.

Reply:
I run the test in https://github.com/rust-lang/rust-clippy/commit/22eeb1109b94ebeae496bacdb9bfb2a3141c5ecc, and I found the `ice-3717.1.fixed` and `ice-3717.2.fixed`.

ice-3717.1.fixed
```
#![deny(clippy::implicit_hasher)]


use std::collections::HashSet;

fn main() {}

pub fn ice_3717<S: ::std::hash::BuildHasher + Default>(_: &HashSet<usize, S>) {
    //~^ ERROR: parameter of type `HashSet` should be generalized over different hashers
    let _ = [0u8; 0];
    let _: HashSet<usize> = HashSet::new();
}
```

ice-3717.2.fixed
```
#![deny(clippy::implicit_hasher)]


use std::collections::HashSet;

fn main() {}

pub fn ice_3717(_: &HashSet<usize>) {
    //~^ ERROR: parameter of type `HashSet` should be generalized over different hashers
    let _ = [0u8; 0];
    let _: HashSet<usize> = HashSet::default();
}
```

However, running it on the current master branch will only generate one test.

Reply:
I don't think this is correct. This lint should only produce one `fixed` file with both locations changed. 2 files should only be emitted, if 2 different suggestions are produced for the same span.

This produces one suggestion for 2 different spans. Maybe the wrong function is used when producing the suggestion? :thinking: